addons/mrp/models/product.py

```diff
@@ -195,3 +205,18 @@ def action_open_quants(self):
             res['context']['single_product'] = False
             res['context'].pop('default_product_tmpl_id', None)
         return res
+
+    def _search_qty_available_new(self, operator, value, lot_id=False, owner_id=False, package_id=False):
+        '''extending the method in stock.product to take into account kits'''
+        product_ids = super(ProductProduct, self)._search_qty_available_new(operator, value, lot_id, owner_id, package_id)
```

Comment:
Could you search for kit on `self` and call `super` on self - kit we do the same thing in this compute
https://github.com/odoo/odoo/blob/e1e6c42d991e79f08725db0a11bfea0de7717a9d/addons/mrp/models/product.py#L170

Reply:
Hello @Whenrow, the fun `_search_qty_available_new` is used to compute a list of products, so when we want to extend it on `mrp` we want to return normal products + all the kits verifying the condition.

To use `_compute_quantities_dict`, if I'm not mistaken, we have to fetch all kits and call it on each kit, and I guess we will be doing the same thing as the pushed code.

Can you correct me please if I'm wrong ? thank you :pray:

Reply:
Don't know if we want to get the product kit and its component in case the kit has a linked quant

Reply:
In this case we don't have a products list, `self` is not used in the query in `_search_qty_available_new`, we search all products matching the domain:

```py
    def _search_qty_available_new(self, operator, value, lot_id=False, owner_id=False, package_id=False):
        ''' Optimized method which doesn't search on stock.moves, only on stock.quants. '''
        product_ids = set()      
        domain_quant = self._get_domain_locations()[0]
        if lot_id:                                                             
            domain_quant.append(('lot_id', '=', lot_id))
        if owner_id:                                          
            domain_quant.append(('owner_id', '=', owner_id))
        if package_id:
            domain_quant.append(('package_id', '=', package_id))                     
        quants_groupby = self.env['stock.quant'].read_group(domain_quant, ['product_id', 'quantity'], ['product_id'], orderby='id')
```

so if we wanted to exclude the kit that has a quant, we would need to do something like:

```diff
+ product_ids = set(product_ids) - set(kit_products)
  for product in kit_products:
```

Reply:
@Whenrow any news about this PR ? Thank you :pray: